From 280c90bb0e49f0d79bd70f7182fa2cf2ba670d39 Mon Sep 17 00:00:00 2001
From: "shuai.zhang" <shuai.zhang@amlogic.com>
Date: Fri, 26 Mar 2021 14:39:42 +0800
Subject: [PATCH] pre_submit_for_google: Camera app adapts to Android R [1/3]

PD#SWPL-44228

Problem:
Android R native Camera application is difficult to operate

Solution:
Camera app adapts to Android R

Verify:
ohm

Signed-off-by: shuai.zhang<shuai.zhang@amlogic.com>
Change-Id: I8386c641f5b23316602f98b77de18c74963eefcd
---
 camera/CameraUtils.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/camera/CameraUtils.cpp b/camera/CameraUtils.cpp
index 67fc116558..37df8848e5 100644
--- a/camera/CameraUtils.cpp
+++ b/camera/CameraUtils.cpp
@@ -52,7 +52,8 @@ status_t CameraUtils::getRotationTransform(const CameraMetadata& staticInfo,
 
     int32_t& flags = *transform;
 
-    bool mirror = (entryFacing.data.u8[0] == ANDROID_LENS_FACING_FRONT);
+    bool mirror = (entryFacing.data.u8[0] == ANDROID_LENS_FACING_FRONT)
+           || (entryFacing.data.u8[0] == ANDROID_LENS_FACING_EXTERNAL);
     int orientation = entry.data.i32[0];
     if (!mirror) {
         switch (orientation) {
-- 
2.29.0

